hedera-node/src/main/java/com/hedera/services/txns/crypto/CryptoApproveAllowanceTransitionLogic.java

```diff
@@ -236,17 +200,14 @@ private void applyFungibleTokenAllowances(final List<TokenAllowance> tokenAllowa
 
 			final var key = FcTokenAllowanceId.from(EntityNum.fromTokenId(tokenId),
 					spender.asEntityNum());
-			if (tokensMap.containsKey(key)) {
-				if (amount == 0) {
-					removeTokenEntity(key, tokensMap, accountToApprove);
-				}
-				// No-Op need to submit adjustAllowance to adjust any allowances
-				continue;
+			if (tokensMap.containsKey(key) && amount == 0) {
+				removeTokenEntity(key, tokensMap, accountToApprove);
+			}
+			if (amount > 0) {
+				tokensMap.put(key, amount);
+				validateAllowanceLimitsOn(accountToApprove, dynamicProperties.maxAllowanceLimitPerAccount());
+				entitiesChanged.put(accountToApprove.getId().num(), accountToApprove);
 			}
-			tokensMap.put(key, amount);
-
-			validateAllowanceLimitsOn(accountToApprove, dynamicProperties.maxAllowanceLimitPerAccount());
-			entitiesChanged.put(accountToApprove.getId().num(), accountToApprove);
 		}
 	}
 
```

Comment:
Similar to above comment. we can remove the method `removeTokenEntity`. all we need is `tokensMap.remove(key);` in ```if (tokensMap.containsKey(key) && amount == 0) {}``` block